src/ensembl/src/content/app/entity-viewer/state/gene-view/entityViewerGeneViewState.ts

```diff
@@ -0,0 +1,55 @@
+export type EntityViewerGeneFunctionState = {
+  selectedTabName: GeneFunctionTabName;
+};
+
+export type EntityViewerGeneRelationshipsState = {
+  selectedTabName: GeneRelationshipsTabName;
+};
+
+export type EntityViewerGeneViewUIState = {
+  selectedGeneTabName: GeneViewTabName;
+  geneFunction: EntityViewerGeneFunctionState;
+  geneRelationships: EntityViewerGeneRelationshipsState;
+};
+
+export type EntityViewerGeneViewState = Readonly<{
+  [genomeId: string]: {
+    [objectId: string]: EntityViewerGeneViewUIState;
```

Comment:
I suggest we use the word "object" as little as possible. Here we know what the objects are — they are genes.

Reply:
That's done